doc/release/upcoming_changes/22541.expired.rst

```diff
@@ -0,0 +1,3 @@
+* The `dtype=` argument to comparison ufuncs is now applied
+  correctly.  This means that only `bool` and `object` are valid
```

Comment:
Small nits:

"That means" , "valid values",  "and `dtype=object` is enforced."